### PR TITLE
fix(styled-react-modal): Fix "export =", modify types, upgrade

### PR DESCRIPTION
**Pull request by hkleungai**
- Modify types.
    - `ModalProps.beforeOpen()` and `ModalProps.beforeClose()` are sync / async callbacks according to doc, instead of being just `Promise<void>` as the current typedef suggests.
    - `ModalProps.onEscapeKeydown()` and `ModalProps.onBackgroundClick()` are modified to accept `KeyboardEvent` and `MouseEvent` parameters respectively.
    - `StyledReactModel.styled()` is modifed to accept whatever args used in `styled.div(...args)`, and return a Model component shape. This behaviour can be observed in [source code](https://github.com/AlexanderRichey/styled-react-modal/blob/60f42fc5bce647387e8e7e8c9e34881f7a63ee0b/src/Modal.jsx#L140-L146).
- Add jsdoc
    - The added jsdoc are based on [author's own API writeup](https://www.npmjs.com/package/styled-react-modal#api).

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

File: types/styled-react-modal/index.d.ts [hunk 1]
```diff
@@ -1,37 +1,79 @@
...
+     * with styles included.
+     */
+    static styled: (
+        ...args: Parameters<ThemedStyledFunctionBase<"div", any>>
+    ) => typeof StyledReactModel;
+}
 
-export default Modal;
-export { BaseModalBackground, ModalProvider };
+export = StyledReactModel;
```
Comment (RyanCavanaugh) [inline types/styled-react-modal/index.d.ts:79]: What happened to the `default` export? Why is this changed to `export =` ? ESM modules should generally not use `export =`.

Comment (hkleungai) [reply types/styled-react-modal/index.d.ts]: Done. I thought it is a cjs module back when I opened the PR. ☹️

**Review by RyanCavanaugh — APPROVED** [review]

**Review by gperl27 — APPROVED** [review]

**Comment by hkleungai** [comment]
Ready to merge
